From f03377c4186090558e1a1a7f0f914c3775af690e Mon Sep 17 00:00:00 2001
From: Richard Zhu <r65037@freescale.com>
Date: Mon, 9 Mar 2009 10:50:39 +0800
Subject: [PATCH] ENGR00108890 [MX37] Some micro sd cards can't be recognized.

Reconfigure the SD/MMC pins' PAD to level up the hw timing
compatible.

Signed-off-by: Richard Zhu <r65037@freescale.com>
---
 arch/arm/mach-mx37/mx37_3stack_gpio.c |   14 ++++++++------
 1 files changed, 8 insertions(+), 6 deletions(-)

diff --git a/arch/arm/mach-mx37/mx37_3stack_gpio.c b/arch/arm/mach-mx37/mx37_3stack_gpio.c
index 0c05df8..a94f73c 100644
--- a/arch/arm/mach-mx37/mx37_3stack_gpio.c
+++ b/arch/arm/mach-mx37/mx37_3stack_gpio.c
@@ -342,6 +342,8 @@ EXPORT_SYMBOL(gpio_activate_audio_ports);
  */
 void gpio_sdhc_active(int module)
 {
+	unsigned int pad_val;
+
 	switch (module) {
 	case 0:
 		mxc_request_iomux(MX37_PIN_SD1_CLK,
@@ -359,37 +361,37 @@ void gpio_sdhc_active(int module)
 
 		mxc_iomux_set_pad(MX37_PIN_SD1_CMD,
 				  PAD_CTL_DRV_MAX | PAD_CTL_DRV_VOT_HIGH |
-				  PAD_CTL_HYS_NONE | PAD_CTL_PUE_PULL |
+				  PAD_CTL_HYS_ENABLE | PAD_CTL_PUE_PULL |
 				  PAD_CTL_47K_PU |
 				  PAD_CTL_ODE_OPENDRAIN_NONE |
 				  PAD_CTL_PKE_ENABLE | PAD_CTL_SRE_FAST);
 		mxc_iomux_set_pad(MX37_PIN_SD1_CLK,
 				  PAD_CTL_DRV_MAX | PAD_CTL_DRV_VOT_HIGH |
-				  PAD_CTL_HYS_NONE | PAD_CTL_PUE_KEEPER |
+				  PAD_CTL_HYS_NONE | PAD_CTL_PUE_PULL |
 				  PAD_CTL_47K_PU |
 				  PAD_CTL_ODE_OPENDRAIN_NONE |
 				  PAD_CTL_PKE_ENABLE | PAD_CTL_SRE_FAST);
 		mxc_iomux_set_pad(MX37_PIN_SD1_DATA0,
 				  PAD_CTL_DRV_MAX | PAD_CTL_DRV_VOT_HIGH |
-				  PAD_CTL_HYS_NONE | PAD_CTL_PUE_KEEPER |
+				  PAD_CTL_HYS_ENABLE | PAD_CTL_PUE_PULL |
 				  PAD_CTL_47K_PU |
 				  PAD_CTL_ODE_OPENDRAIN_NONE |
 				  PAD_CTL_PKE_ENABLE | PAD_CTL_SRE_FAST);
 		mxc_iomux_set_pad(MX37_PIN_SD1_DATA1,
 				  PAD_CTL_DRV_MAX | PAD_CTL_DRV_VOT_HIGH |
-				  PAD_CTL_HYS_NONE | PAD_CTL_PUE_KEEPER |
+				  PAD_CTL_HYS_ENABLE | PAD_CTL_PUE_PULL |
 				  PAD_CTL_47K_PU |
 				  PAD_CTL_ODE_OPENDRAIN_NONE |
 				  PAD_CTL_PKE_ENABLE | PAD_CTL_SRE_FAST);
 		mxc_iomux_set_pad(MX37_PIN_SD1_DATA2,
 				  PAD_CTL_DRV_MAX | PAD_CTL_DRV_VOT_HIGH |
-				  PAD_CTL_HYS_NONE | PAD_CTL_PUE_KEEPER |
+				  PAD_CTL_HYS_ENABLE | PAD_CTL_PUE_PULL |
 				  PAD_CTL_47K_PU |
 				  PAD_CTL_ODE_OPENDRAIN_NONE |
 				  PAD_CTL_PKE_ENABLE | PAD_CTL_SRE_FAST);
 		mxc_iomux_set_pad(MX37_PIN_SD1_DATA3,
 				  PAD_CTL_DRV_MAX | PAD_CTL_DRV_VOT_HIGH |
-				  PAD_CTL_HYS_NONE | PAD_CTL_PUE_PULL |
+				  PAD_CTL_HYS_ENABLE | PAD_CTL_PUE_PULL |
 				  PAD_CTL_47K_PU |
 				  PAD_CTL_ODE_OPENDRAIN_NONE |
 				  PAD_CTL_PKE_ENABLE | PAD_CTL_SRE_FAST);
-- 
1.5.4.4

